cmd/peer-rest-client.go

```diff
@@ -0,0 +1,418 @@
+/*
+ * Minio Cloud Storage, (C) 2019 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/gob"
+	"io"
+	"net/url"
+
+	"github.com/minio/minio/cmd/http"
+	"github.com/minio/minio/cmd/logger"
+	"github.com/minio/minio/cmd/rest"
+	"github.com/minio/minio/pkg/event"
+	xnet "github.com/minio/minio/pkg/net"
+	"github.com/minio/minio/pkg/policy"
+)
+
+// Abstracts a remote disk.
+type peerRESTClient struct {
+	host       *xnet.Host
+	restClient *rest.Client
+	connected  bool
+	lastError  error
+	instanceID string // REST server's instanceID which is sent with every request for validation.
```

Comment:
connected, lastError and instanceID are not needed for peerRESTClient

Reply:
Kept `connected` and currently retrying in `call` based on `connected and removed lastError and instanceID like we talked about.